src/libraries/System.Private.CoreLib/gen/ProductVersionInfoGenerator.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Reflection;
+using Microsoft.CodeAnalysis;
+
+namespace Generators
+{
+    [Generator]
+    public partial class ProductVersionInfoGenerator : IIncrementalGenerator
+    {
+        public void Initialize(IncrementalGeneratorInitializationContext context)
+        {
+            context.RegisterPostInitializationOutput(ctx =>
+            {
+                string? informationalVersion = typeof(ProductVersionInfoGenerator).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;
```

Comment:
I'm not sure how our build system handles this. Is this guaranteed to always match what typeof(object).Assembly would in the built corelib?

Reply:
As far as I can tell, the build system stamps the exact same `AssemblyInformationalVersionAttribute` on all binaries.